test/test_sparse_csr.py

```diff
@@ -0,0 +1,174 @@
+import torch
+
+torch.set_default_dtype(torch.double)
+
+import functools
+import random
+import operator
+import numpy as np
+import warnings
+from torch.testing._internal.common_utils import TestCase, run_tests, load_tests
+
+# load_tests from torch.testing._internal.common_utils is used to automatically filter tests for
+# sharding on sandcastle. This line silences flake warnings
+load_tests = load_tests
+
+class TestSparseCSR(TestCase):
+    def gen_sparse_csr(self, shape, nnz):
+        total_values = functools.reduce(operator.mul, shape, 1)
+        dense = np.random.randn(total_values)
+        fills = random.sample(list(range(total_values)), total_values - nnz)
+
+        for f in fills:
+            dense[f] = 0
+        dense = torch.from_numpy(dense.reshape(shape))
+
+        return dense.to_sparse_csr()
+
+    def setUp(self):
+        # These parameters control the various ways we can run the test.
+        # We will subclass and override this method to implement CUDA
+        # tests
+        self.is_cuda = False
+        self.device = 'cpu'
```

Comment:
What about tests with coalesced sparse tensors ?

Reply:
CSR layout does not support coalesced/uncoalesced tensors ATM.

Reply:
CSR format does not support repeated indices, so I think supporting uncoaleasced tensors would not be possible. CSR tensors are coalesced by definition. However, converting uncoaleasced COO to CSR would require a test, but only after the direct COO->CSR method is implemented.